src/drivers/adc/adc.c

```diff
@@ -0,0 +1,88 @@
+#include "FreeRTOS.h"
+#include "drivers/adc/adc.h"
+#include "stm32h7xx_hal.h"
+#include <semphr.h>
+
+extern ADC_HandleTypeDef hadc1;
+
+SemaphoreHandle_t adc_conversion_semaphore;
+SemaphoreHandle_t adc_semaphore;
+
+adc_channel_t current_adc_channel = PROCESSOR_BOARD_VOLTAGE;
+
+#define ADC_MAX_COUNTS 65535
+
+w_status_t adc_init(void)
+{
+    adc_conversion_semaphore = xSemaphoreCreateBinary();
+    adc_semaphore = xSemaphoreCreateBinary();
+
+    if (adc_semaphore == NULL || adc_conversion_semaphore == NULL)
```

Comment:
```suggestion
    if ((NULL == adc_semaphore) || (NULL == adc_conversion_semaphore))
```
as per BARR-C:2018 1.4b and 8.6a